### PR TITLE
giggle: fix cross, crash

### PR DESCRIPTION
**Pull request by sgn**
Close #27151

@ndowens

### CODE REVIEW

**Comment by ndowens** [comment]
Hmm how does the cross patch work to fix? What seems to be doing is removing stuff related to time?

**Comment by sgn** [comment]
> Hmm how does the cross patch work to fix? What seems to be doing is removing stuff related to time?

By not depends on `configure`'s result but libc's pre-defined macro. And not trying to mess to `sizeof(time_t)`, (messing like that is not correct anyway).

File: srcpkgs/giggle/patches/cross.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+-#if STRPTIME_HAS_GNU
++#ifdef __GLIBC__
+ 	const gchar *returned;
+ 	struct tm   *tm;
+ 
+@@ -145,9 +145,11 @@ git_revisions_get_time (const gchar *date)
+ 	return tm;
+ #else
+ 	struct tm *tm = g_new0 (struct tm, 1);
++	long       ltime;
```
Comment (ericonr) [inline srcpkgs/giggle/patches/cross.patch:53]:
```
#include <stdio.h>

int main()
{
        printf("%ld\n", sizeof(long));
}
```

on my rpi prints 4, so I think it'd be more correct to use `long long` here.

Comment (sgn) [reply srcpkgs/giggle/patches/cross.patch:53]: Make sense, the original code is cursed.

Comment (sgn) [reply srcpkgs/giggle/patches/cross.patch]: Anyway, correct format specifier for `size_t` is `zu` :-p

**Comment by sgn** [comment]
https://gitlab.gnome.org/GNOME/giggle/-/merge_requests/3

File: srcpkgs/giggle/patches/cross.patch (hunk 1 above)
Comment (ericonr) [reply srcpkgs/giggle/patches/cross.patch]: Heh, true

**Pull request merged**